src/contracts/oracle-proxy/AggregatorV3Proxy.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+//**************************************************************************************************
+//                                      CHAINLINK
+//**************************************************************************************************
+import {AggregatorV3Interface} from "@chainlink/shared/interfaces/AggregatorV2V3Interface.sol";
+import {IDIAOracleV2} from "src/interfaces/oracles/IDIAOracleV2.sol";
+
+contract AggregatorV3Proxy is AggregatorV3Interface {
+    string public pairSymbol;
+    IDIAOracleV2 public aggregator;
+
+    error AggregatorV3Proxy__InvalidData();
+
+    constructor(address _aggregator, string memory _pairSymbol) {
+        if (_aggregator == address(0) || bytes(_pairSymbol).length == 0) {
+            revert AggregatorV3Proxy__InvalidData();
+        }
+        aggregator = IDIAOracleV2(_aggregator);
+        pairSymbol = _pairSymbol;
+    }
+
+    function decimals() external pure override returns (uint8) {
+        return 8;
```

Comment:
Let's include this in the constructor so we can use it for others in the future without any change